counsel.el

```diff
@@ -2699,6 +2719,34 @@ Note: don't use single quotes for the regex."
 (ivy-set-occur 'counsel-rg 'counsel-ag-occur)
 (ivy-set-display-transformer 'counsel-rg 'counsel-git-grep-transformer)
 
+(defcustom counsel-rg-pcre-flag 'auto
+  "PCRE flag used by Ripgrep.
+
+If `auto', flag will be determined automatically.
+
+If `nil', PCRE support is disabled unless \"--pcre2\" or \"-P\"
+is part of `counsel-rg-base-command'.
+
+If a string, pass value to Ripgrep when PCRE support is
+requested.
```

Comment:
Sorry, this sentence isn't clear - when, how, and by whom is PCRE support requested? Did you mean something like the following? (Feel free to improve on it in a different way.)
```el
If a string, it should comprise Ripgrep switch(es) which enable
PCRE support.
```